src/main/java/org/opensearch/ad/rest/RestCreateDetectorAction.java

```diff
@@ -1,134 +1,241 @@
+    private NamedXContentRegistry.Entry registerAggregation(SearchPlugin.AggregationSpec spec) {
+        return new NamedXContentRegistry.Entry(BaseAggregationBuilder.class, spec.getName(), (p, c) -> {
+            String name = (String) c;
+            return spec.getParser().parse(p, name);
+        });
     }
 
-    private ExtensionRestResponse handlePutRequest(ExtensionRestRequest request) {
-        String name = request.param("name");
+    private CreateIndexRequest initAnomalyDetectorIndex() throws FileNotFoundException {
+        JsonpMapper mapper = sdkClient._transport().jsonpMapper();
+        ((JacksonJsonpMapper) mapper).objectMapper().registerModule(new JavaTimeModule());
+        JsonParser parser = null;
         try {
-            worldName = URLDecoder.decode(name, StandardCharsets.UTF_8);
-        } catch (IllegalArgumentException e) {
-            return new ExtensionRestResponse(request, BAD_REQUEST, e.getMessage());
+            parser = mapper
+                .jsonProvider()
+                .createParser(new ByteArrayInputStream(getAnomalyDetectorMappings().getBytes(StandardCharsets.UTF_8)));
+        } catch (Exception e) {
+            e.printStackTrace();
         }
-        return new ExtensionRestResponse(request, OK, "Updated the world's name to " + worldName);
-    }
 
-    private ExtensionRestResponse handleBadRequest(ExtensionRestRequest request) {
-        return new ExtensionRestResponse(request, NOT_FOUND, "Extension REST action improperly configured to handle " + request.toString());
+        CreateIndexRequest request = null;
+        try {
+            request = new CreateIndexRequest.Builder()
+                .index(ANOMALY_DETECTORS_INDEX)
+                .mappings(TypeMapping._DESERIALIZER.deserialize(parser, mapper))
+                .build();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return request;
     }
 
-    private String parseJsonAdjective(String json) {
-        // TODO: Once CreateComponents has an XContentRegistry available we can parse from there
-        // For now we just hack our way into the result.
-        boolean foundLabel = false;
-        boolean foundColon = false;
-        for (String s : json.split("\"")) {
-            if (!foundLabel) {
-                foundLabel = "adjective".equals(s);
-            } else if (!foundColon) {
-                foundColon = s.contains(":");
-            } else {
-                // This is the adjective!
-                return s;
+    @Override
+    public ExtensionRestResponse handleRequest(ExtensionRestRequest request) {
+        if (!EnabledSetting.isADPluginEnabled()) {
+            throw new IllegalStateException(CommonErrorMessages.DISABLED_ERR_MSG);
+        }
+        Method method = request.method();
+
+        NamedXContentRegistry xContentRegistry = new NamedXContentRegistry(getNamedXWriteables());
+        XContentParser parser = null;
+        try {
```

Comment:
I think we can try to wrap most of this handler method in a try/catch, where the catch returns an ExtensionRestResponse failure with the exception details.